modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/opentofu/callbacks/OpenTofuDeploymentResultCallbackManager.java

```diff
@@ -88,8 +88,33 @@ public void deployCallback(UUID taskId, OpenTofuResult result) {
     /**
      * Callback method after the modification task is completed.
      */
-    public void modifyCallback(UUID taskId, OpenTofuResult openTofuResult) {
-        handleCallbackOpenTofuResult(taskId, openTofuResult, DeploymentScenario.MODIFY);
+    public void modifyCallback(UUID serviceId, OpenTofuResult openTofuResult) {
+        handleCallbackOpenTofuResult(serviceId, openTofuResult, DeploymentScenario.MODIFY);
+        updateModificationAuditWithResult(serviceId, openTofuResult);
+    }
+
+    private void updateModificationAuditWithResult(UUID serviceId, OpenTofuResult result) {
+        ServiceModificationAuditEntity auditEntityInProgress;
+        if (Objects.isNull(result.getRequestId())) {
+            ServiceModificationAuditEntity entity = new ServiceModificationAuditEntity();
+            entity.setServiceId(serviceId);
+            entity.setTaskStatus(TaskStatus.IN_PROGRESS);
```

Comment:
the code is fine. Just for understanding, does this case really happen? Because callback is executed only when successful or failure.

Reply:
Removed the code for this case.